test/functional/tool_wallet.py

```diff
@@ -36,8 +42,32 @@ def assert_tool_output(self, output, *args):
         assert_equal(stderr, '')
         assert_equal(stdout, output)
 
-    def run_test(self):
+    def wallet_path(self):
+        return os.path.join(self.nodes[0].datadir, 'regtest', 'wallets', 'wallet.dat')
+
+    def wallet_shasum(self):
+        h = hashlib.sha1()
+        mv = memoryview(bytearray(BUFFER_SIZE))
+        with open(self.wallet_path(), 'rb', buffering=0) as f:
+            for n in iter(lambda : f.readinto(mv), 0):
+                h.update(mv[:n])
+        return h.hexdigest()
+
+    def wallet_timestamp(self, op):
+        timestamp = os.path.getmtime(self.wallet_path())
+        self.log.info('Wallet file timestamp {0}: {1}'.format(op, timestamp))
+        return timestamp
 
+    def log_wallet_permissions(self):
+        result = oct(os.lstat(self.wallet_path()).st_mode)[-3:]
+        self.log.info('Verifying wallet file permissions: {}'.format(result))
```

Comment:
You aren't verifying anything here, it seems. I'd prefer if this was just a simple getter util method and the caller logs/asserts if appropriate.

Reply:
Done.